internal/operator/reconciliations/api_gateway/reconciliation.go

```diff
@@ -0,0 +1,149 @@
+package api_gateway
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/util/retry"
+
+	operatorv1alpha1 "github.com/kyma-project/api-gateway/apis/operator/v1alpha1"
+	"github.com/kyma-project/api-gateway/internal/described_errors"
+	"github.com/kyma-project/api-gateway/internal/operator/resources"
+	networkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ApiGatewayReconciliation interface {
+	Reconcile(ctx context.Context, apiGatewayCR operatorv1alpha1.APIGateway, apiGatewayResourceListPath string) (operatorv1alpha1.APIGateway, described_errors.DescribedError)
+}
+
+type Reconciliation struct {
+	Client client.Client
+}
+
+const (
+	reconciliationFinalizer string = "apigateways.operator.kyma-project.io/api-gateway-reconciliation"
+	dafaultGatewayName      string = "kyma-gateway"
```

Comment:
There is a typo `dafault` . Also I think we should expose this variables from the gateway package from PR https://github.com/kyma-project/api-gateway/pull/537 and use it here.

Reply:
Good catch, yes, it would be best to have it in gateway package and reuse it from there.